security/certs.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package security
+
+import (
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"os"
+	"path"
+
+	"github.com/cockroachdb/cockroach/server"
+	"github.com/cockroachdb/cockroach/util"
+)
+
+// writeCertificateAndKey takes a x509 certificate and key and writes
+// them out to the individual files.
+// The certificate is written to <prefix>.crt and the key to <prefix>.key.
+// TODO(marc): figure out how to include the plaintext certificate in the .crt file.
```

Comment:
do we care about the permissions? The directory will be `0755`, but the private keys probably should not. Not that it would matter in most deployments these days..